pkg/controllers/managementuser/rbac/namespace_handler.go

```diff
@@ -567,3 +568,40 @@ func updateStatusAnnotation(hasPRTBs bool, namespace *v1.Namespace, mgr *manager
 	}
 
 }
+
+// asyncCleanupRBAC will wait for a Terminating namespace to be fully deleted before removing the associated RBAC.
+func (n *nsLifecycle) asyncCleanupRBAC(namespaceName string) {
+	go func() {
+		backoff := wait.Backoff{
+			Duration: 5 * time.Second,
+			Factor:   2.0,
+			Jitter:   0.1,
+			Steps:    10,
+			Cap:      5 * time.Minute,
+		}
+
+		err := wait.ExponentialBackoff(backoff, func() (bool, error) {
+			_, err := n.m.nsLister.Get("", namespaceName)
+			if err != nil {
+				if apierrors.IsNotFound(err) {
+					// Namespace is fully deleted, clean up RBAC
+					err := n.reconcileNamespaceProjectClusterRole(&v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespaceName}})
+					if err != nil {
+						logrus.Errorf("error cleaning up RBAC for namespace %s: %v", namespaceName, err)
+						return false, err
```

Comment:
My understanding is this change is for waiting until the namespace is terminated to remove the RBAC. If we return false here, it's going to repeatedly try to do `reconcileNamespaceProjectClusterRole` after the namespace is deleted even if that fails. My worry is that if something goes wrong we're just flooding the logs. Maybe we should only attempt this once? If you think it's a good idea to attempt it multiple times, I don't feel too strongly about it.
```suggestion
						return true, err
```

Reply:
Reasonable thought.  Probably ok to return true here since the underlying error isn't likely to be solved by trying again.  I'll update it.